zio-http/jvm/src/main/scala/zio/http/netty/client/ClientInboundHandler.scala

```diff
@@ -35,6 +35,7 @@ private[netty] final class ClientInboundHandler(
   jReq: HttpRequest,
   onResponse: Promise[Throwable, Response],
   onComplete: Promise[Throwable, ChannelState],
+  onFailure: Promise[Nothing, Throwable],
```

Comment:
How is this `Promise` used? I can't find any usage of `await` or `poll` on it, so AFAICT we're only completing it without really using it?